locale/tests/locale.cc

```diff
@@ -0,0 +1,70 @@
+#include <catch.hpp>
+#include <leatherman/locale/locale.hpp>
+#include <boost/locale.hpp>
+
+using namespace std;
+using namespace leatherman::locale;
+
+#define _(x) x
+static auto literal = _("requesting {1,number}.");
+static auto proj = "leatherman_test";
+
+SCENARIO("a default locale") {
+    auto loc = get_locale("", proj);
+    formatter fmt(proj);
+
+    GIVEN("std::ostream") {
+        stringstream ss;
+        ss.imbue(loc);
+        THEN("messages should not be translated") {
+            ss << boost::locale::translate(literal);
+            REQUIRE(ss.str() == literal);
+        }
+    }
+
+    GIVEN("leatherman::locale::translate") {
+        THEN("messages should not be translated") {
+            REQUIRE(translate(literal, proj) == literal);
+        }
+    }
+
+    GIVEN("leatherman::locale::format") {
+        THEN("messages should not be translated") {
+            REQUIRE(fmt(literal, 1.25) == "requesting 1.25.");
+        }
+    }
+
+    clear_domain(proj);
+}
+
+SCENARIO("a french locale") {
+    auto loc = get_locale("fr.UTF-8", proj);
+    formatter fmt(proj);
+
+    GIVEN("std::ostream") {
+        stringstream ss;
+        ss.imbue(loc);
+        THEN("messages should be translated") {
+            ss << boost::locale::translate(literal);
+            REQUIRE(ss.str() == "demande {1,number}.");
+        }
+    }
+
+    GIVEN("leatherman::locale::translate") {
+        THEN("messages should be translated") {
+            REQUIRE(translate(literal, proj) == "demande {1,number}.");
+        }
+    }
+
+    GIVEN("leatherman::locale::format") {
+        THEN("messages should be translated") {
+#if defined(__MAC_OS_X_VERSION_MIN_REQUIRED) || defined(_WIN32)
```

Comment:
Mac Boost says its built with icu, and I've written tests explicitly asking for icu. So I'm a bit confused.

Minimal test-case
```
#include <boost/locale.hpp>

int main(int argc, char** argv)
{
    boost::locale::format fmt("{1,number} should be 1,25.");
    fmt % 1.25;
    boost::locale::generator gen;
    std::cout << fmt.str(gen("fr-fr.UTF-8")) << std::endl;
    return 0;
}
```